og_access/og_access.test

```diff
@@ -345,6 +345,59 @@ class OgAccessTestCase extends DrupalWebTestCase {
 
     $this->assertFalse(node_access('view', $node, $user2), 'Group member can not view unpublished node.');
   }
+
+  /**
+   * Test unpublished node.
+   */
+  function testUnpublishedGroupAccess() {
+    $user1 = $this->user1;
+    $user2 = $this->user2;
+    $this->drupalLogin($user1);
+
+    // Create a group node and set as private.
+    $settings = array();
+    $settings['type'] = $this->group_type;
+    $settings[OG_GROUP_FIELD][LANGUAGE_NONE][0]['value'] = 1;
+    $settings[OG_ACCESS_FIELD][LANGUAGE_NONE][0]['value'] = 1;
+    $group_node = $this->drupalCreateNode($settings);
+
+    // Add another user to group.
+    og_group('node', $group_node->nid, array('entity' => $user2));
+    drupal_static_reset('node_access');
+
+    $this->assertTrue(entity_access('view', 'node', $group_node, $user2), 'Group member can view published group.');
+
+    // Get user's groups he has access to using og_get_entity_groups().
+    $groups1 = og_get_entity_groups('user', $user2, array(OG_STATE_ACTIVE), NULL, TRUE);
+    // Get user's groups he has access to using og_get_groups_by_user().
+    $groups2 = og_get_groups_by_user($user2, 'node', TRUE);
+
+    $group1 = !empty($groups1['node']) ? in_array($group_node->nid, $groups1['node']) : FALSE;
+    $group2 = !empty($groups2) ? in_array($group_node->nid, $groups2) : FALSE;
+
+    $this->assertTrue($group1, 'Group member is getting published group from og_get_entity_groups().');
+    $this->assertTrue($group2, 'Group member is getting published group from og_get_groups_by_user().');
+
+    $group_node->status = NODE_NOT_PUBLISHED;
+    node_save($group_node);
+    drupal_static_reset('node_access');
+
+    $this->assertFalse(entity_access('view', 'node', $group_node, $user2), "Group member can't view unpublished group.");
+
+    // Clear the cache because permissions were changed.
+    drupal_static_reset('og_get_entity_groups');
+
+    // Get user's groups he has access to using og_get_entity_groups().
```

Comment:
`groups he has ..` => `groups they have ..` (here and below)

because they can be a girl to -- like in your example :)